### PR TITLE
Make json schema leader

### LINKED ISSUE #17: Port config to json schema
Now the schema is defined in zod. Would be nicer to have the JSON schema and use [json-schema-to-zod](https://www.npmjs.com/package/json-schema-to-zod) instead of vice-versa.

We could add:

* Title
* Units 
* zod.describe --> description

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #17 

TODO
- [x] convert zod schema to json schema
- [x] derive type from json schema
- [x] use ajv to validate
- [x] use ajv to parse
- [ ] pimp json schema
  - [x] rename descriptions to titles for leafs
  - [x] add titles to type=object
  - [x] add descriptions, (in Markdown?)
    - [x] add description to theta_0 to test rendering
  - [x] add min/max
    - [x] add min to theta_0 to test error reporting
  - [x] use in app form generation
- [x] use ajv validate
  - [x] in package
  - [x] in app
- [x] document type generation from json schema. When, how
- [x] drop zod
  - [x] package
  - [x] app
- [x] have partial and concrete Config, Concrete has all values filled with user supplied or default. Concrete can be use by class model and partial can be used in app.
- [x] add config.json to exports in package.json
- [x] make diff viewer show latest configs, now is caching old version
- [x] second edit of a number field  on download is a string

### CODE REVIEW

**Comment by sverhoeven** [comment]
To not have to call the json2ts command we could create a typescript plugin that generates a dts file when config.json is changed.

See for example https://github.com/remix-run/react-router/blob/8fc8239ff6889b9d0f7ed4aba06b74daa61203bb/packages/react-router-dev/typescript/plugin.ts#L11 and https://github.com/remix-run/react-router/blob/dev/.changeset/typesafety.md#typescript-plugin

**Comment by Peter9192** [comment]
This branch currently doesn't pass my test as described in https://github.com/classmodel/class-web/issues/50.

After updating the reference of the duplicated experiment, the permutation doesn't seem to update. 

Repeatedly editing the reference and permution of the original experiment also seems to sometimes get confused.

File: apps/class-solid/src/components/ajvForm.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      const path = key.split(".");
+      while (path.length > 0) {
+        path.pop();
+        const parentKey = path.join(".");
+        if (parentKey) {
+          formErrors[parentKey] = "Child properties has one or more errors";
+        }
+      }
+    }
+    return formErrors as FormErrors<TFieldValues>;
```
Comment (Peter9192) [inline apps/class-solid/src/components/ajvForm.tsx:41]: This is great, but the formatting is a bit weird. Notice how the title of the next field is suddenly showing up when there's an error.

image

File: packages/class/src/config.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      "properties": {
+        "h_0": {
+          "type": "number",
+          "default": 200,
+          "title": "ABL height [m]"
+        },
+        "theta_0": {
+          "type": "number",
+          "default": 288,
+          "title": "Mixed-layer potential temperature [K]",
```
Comment (Peter9192) [inline packages/class/src/config.json:15]: Would be nice to have units as a separate attribute

Comment (sverhoeven) [reply packages/class/src/config.json:15]: Created #52 for it

File: apps/class-solid/src/components/ajvForm.tsx (hunk 1 above)
Comment (sverhoeven) [reply apps/class-solid/src/components/ajvForm.tsx:41]: localhost_3000_ (4)

I moved the error closer to the input field. Does this help?

**Pull request merged**